### PR TITLE
chore: update code owner details

### PR DESCRIPTION
**Pull request by zainab-amir**
**Description:** Updates code owner info from 2u-vanguards to 2u-infinity

**JIRA:** https://2u-internal.atlassian.net/browse/VAN-1857

**Reviewers:**
- [x] tag reviewer @feanil 
- [x] tag reviewer @openedx/2u-infinity 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green

### CODE REVIEW

**Comment by zainab-amir** [comment]
Unable to add @openedx/2u-infinity as reviewers to this PR. Related issue created for axim https://github.com/openedx/axim-engineering/issues/1065

**Review by hurtstotouchfire — APPROVED** [review]

File: catalog-info.yaml [hunk 1]
```diff
@@ -13,6 +13,6 @@ metadata:
   annotations:
     openedx.org/arch-interest-groups: ""
 spec:
-  owner: group:vanguards
+  owner: group:edx-infinity
```
Comment (hurtstotouchfire) [inline catalog-info.yaml]: I thought that we changed all these legacy group names to 2u-something but Asad can fix it if this is wrong.

**Comment by arbrandes** [comment]
I added the `2u-infinity` team to this repository with explicit write permissions, after which it became possible to add it as a reviewer.

File: catalog-info.yaml (hunk 1 above)
Comment (feanil) [reply catalog-info.yaml]: Agreed, this should be `2u-infinity` and if that team does not have the right people on it, we can update but it should be the team used moving forward.

**Review by feanil — CHANGES_REQUESTED** [review]

**Comment by asadazam93** [comment]
@feanil can you make me (asadazam93) the maintainer of https://github.com/orgs/openedx/teams/2u-infinity? The team members are not up to date

**Review by asadazam93 — APPROVED** [review]

**Review by feanil — APPROVED** [review]

**Pull request merged**